parser-typechecker/src/Unison/TermPrinter.hs

```diff
@@ -216,7 +220,7 @@ pretty0
                     in uses $ [pretty0 n (ac 0 Block im' doc) tm]
     App' x (Constructor' DD.UnitRef 0) ->
       paren (p >= 11) $ (fmt S.DelayForceChar $ l "!") <> pretty0 n (ac 11 Normal im doc) x
-    LamNamed' v x | (Var.name v) == "()" ->
+    Delay' x  ->
```

Comment:
Keying off the variable name is a little fragile. Now we're using the `Delay'` pattern, which treats any unary lambda that ignores its argument as a delay.